File: Samples/Android/app/src/main/java/com/tencent/voiceai/demo/viewmodel/SettingViewModel.kt
// Copyright (c) 2026 Tencent. All rights reserved.
// VoiceAI Demo - 设置 ViewModel

package com.tencent.voiceai.demo.viewmodel

import androidx.lifecycle.ViewModel

/**
 * 设置页面的 ViewModel。
 *
 * 设置页目前仅保留「默认音色」「声纹注册」「版本」等展示 / 跳转项，
 * 其状态由 TtsVoiceSettingViewModel 提供，本类暂不持有开关状态。
 * 后续若新增设置项，可在此以 mutableStateOf + private set 对外只读发布。
 */
class SettingViewModel : ViewModel()

File: Samples/iOS/VoiceAIKitDemo/Demo/ViewModels/SettingViewModel.swift
// Copyright (c) 2026 Tencent. All rights reserved.
// VoiceAI Demo - 设置页 ViewModel (对齐 Android SettingViewModel.kt)

import Foundation

final class SettingViewModel: ObservableObject {
    @Published var voiceDisplayName: String = ""
    @Published var denoiseMode: DenoiseMode = VoiceAIPrefs.shared.realtimeDenoiseMode
    @Published var hasVoiceprint: Bool = false

    func refresh() {
        let voiceId = VoiceAIKit.shared.getParams()?.ttsVoiceId ?? VoiceAIKit.defaultTtsVoiceId
        voiceDisplayName = TtsVoiceCatalog.displayName(of: voiceId)
        denoiseMode = VoiceAIPrefs.shared.realtimeDenoiseMode
        hasVoiceprint = VoiceprintStore.hasVoiceprint
    }

    func setDenoise(_ mode: DenoiseMode) {
        denoiseMode = mode
        VoiceAIPrefs.shared.realtimeDenoiseMode = mode
    }

    /// App 版本号。
    var appVersion: String {
        let v = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? "1.0"
        let b = Bundle.main.infoDictionary?["CFBundleVersion"] as? String ?? "1"
        return "\(v) (\(b))"
    }
}
